route/build/cpl/RtmMod.F90

```diff
@@ -811,29 +811,41 @@ SUBROUTINE restFile_read_pfile( pnamer )
     USE RtmFileUtils, ONLY: relavu, getavu, opnfil
     USE RtmVar,       ONLY: inst_suffix
     USE public_var,   ONLY: rpntfil
+    USE globalData,   ONLY: simDatetime
 
     implicit none
     ! Arguments:
     character(len=*), intent(out) :: pnamer ! full path of restart file
     ! Local variables:
-    integer :: i                  ! indices
-    integer :: nio                ! restart unit
-    integer :: status             ! substring check status
-    character(len=256) :: locfn   ! Restart pointer file name
+    integer :: i                    ! indices
+    integer :: nio                  ! restart unit
+    integer :: sec_in_day           ! time in second
+    character(len=17) :: timestamp  ! datetime string in yyyy-mm-dd-sssss
+    integer :: status               ! substring check status
+    character(len=256) :: locfn     ! Restart pointer file name
     !--------------------------------------------------------
 
     if (masterproc) then
       write(iulog,*) 'Reading restart pointer file....'
     endif
 
-    nio = getavu()
-    locfn = './'// trim(rpntfil)//trim(inst_suffix)
+    nio = getavu() ! get available unit number
```

Comment:
Note, that new FORTRAN constructs allow you to NOT need getavu() anymore. There's a way to open a file with an available unit.

This would be a good modernization effort to do. But, not completely required.

Reply:
I think the only thing in RtmFileUtils that is maybe useful is opnfil, but the use of getavu and relavu could be removed and handled within opnfil.

Reply:
Thanks for suggestions on ascii file handling. I will clean this up with another PR. This RtmFileUtils and ascii_utils could be merged and several routines (getavu and relavu, getSpareUnit) can be removed.